ArduPlane/reverse_thrust.cpp

```diff
@@ -17,18 +17,25 @@
  */
 #include "Plane.h"
 
+// Check USE_REV_THRUST paramater for the given reverse thrust option
+bool Plane::reverse_thrust_enabled(UseReverseThrust use_reverse_thrust_option) const
+{
+    return (g.use_reverse_thrust.get() & int32_t(use_reverse_thrust_option)) != 0;
+}
+
 /*
   see if reverse thrust should be allowed in the current flight state
  */
 bool Plane::allow_reverse_thrust(void) const
 {
-    // check if we should allow reverse thrust
-    bool allow = false;
-
-    if (g.use_reverse_thrust == USE_REVERSE_THRUST_NEVER || !have_reverse_thrust()) {
```

Comment:
This early return is the cause of the bug. It means we never get to the default case in the switch for "other modes".